### PR TITLE
USWDS-Site: Add changelog for date picker aria-disabled updates [#6013]

### PR DESCRIPTION
**Pull request by RachelCorsino**
# Summary

**This PR provides a changelog entry for the updates made to the date picker component.** The `readonly` attribute has been added to `aria-disabled` to mimic the disable behaviors while remaining discoverable for accessibility purposes.

## Related issue

[USWDS - Date Picker: Updated toggleCalendar function to account for aria-disabled attribute + readonly #6013](https://github.com/uswds/uswds/pull/6013)

## Preview link

- [USWDS-Site: Date Picker](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/rc-changelog-datepicker-6013/components/date-picker/)
- [USWDS-Site: Date range picker](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/rc-changelog-datepicker-6013/components/date-range-picker/)

## Testing and review

1. Ensure changelog entry shows up under "Latest Updates" on Date Picker component page.

### CODE REVIEW

File: _data/changelogs/component-date-picker.yml [hunk 1]
```diff
@@ -2,6 +2,15 @@ title: Date picker
 type: component
 changelogURL:
 items:
+  - date: NNNN-NN-NN
+    summary: Added readonly attribute to aria-disabled.
+    summaryAdditional: Now, the input element and calendar icon remain discoverable but disabled behaviors are replicated.
```
Comment (mahoneycm) [inline _data/changelogs/component-date-picker.yml:7]: **suggestion:** Since the change covers more than just adding the `readonly` attribute we could maybe reword to something along the lines of:

```suggestion
    summary: Enhanced `aria-disabled` behaviors to match `disabled` behaviors.
    summaryAdditional: Now, the input element and calendar icon remain discoverable to screen readers but cannot be interacted with.
```

Feel free to toy with it!

File: _data/changelogs/component-date-picker.yml [hunk 2]
```diff
@@ -2,6 +2,15 @@ title: Date picker
...
+    affectsAccessibility: true
+    affectsJavascript: true
+    affectsMarkup: true
```
Comment (mahoneycm) [inline _data/changelogs/component-date-picker.yml:10]: Since the markup changes come from the JS, and are not something the user has to manually add to their markup, we can remove this line.

```suggestion

```

File: _data/changelogs/component-date-picker.yml [hunk 3]
```diff
@@ -2,6 +2,15 @@ title: Date picker
...
+    githubPr: #6013
```
Comment (mahoneycm) [inline _data/changelogs/component-date-picker.yml:11]: We should remove the `#` here.

It's causing changelog to not include the link to the PR

image

```suggestion
    githubPr: 6013
```

**Review by mahoneycm — CHANGES_REQUESTED** [review]

File: _data/changelogs/component-date-picker.yml (hunk 1 above)
Comment (RachelCorsino) [reply _data/changelogs/component-date-picker.yml]: Resolved in [fefb694](https://github.com/uswds/uswds-site/pull/2794/commits/fefb6949907954544486e519967cf3c9c5b7f530)

File: _data/changelogs/component-date-picker.yml (hunk 2 above)
Comment (RachelCorsino) [reply _data/changelogs/component-date-picker.yml]: Resolved in [fefb694](https://github.com/uswds/uswds-site/pull/2794/commits/fefb6949907954544486e519967cf3c9c5b7f530)

File: _data/changelogs/component-date-picker.yml (hunk 3 above)
Comment (RachelCorsino) [reply _data/changelogs/component-date-picker.yml]: Resolved in [fefb694](https://github.com/uswds/uswds-site/pull/2794/commits/fefb6949907954544486e519967cf3c9c5b7f530)

File: _data/changelogs/component-date-picker.yml [hunk 4]
```diff
@@ -2,6 +2,14 @@ title: Date picker
 type: component
 changelogURL:
 items:
+  - date: NNNN-NN-NN
+    summary: Enhanced `aria-disabled` behaviors to match `disabled` behaviors.
+    summaryAdditional: Now, the input element and calendar icon remain discoverable to screen readers but cannot be interacted with.
```
Comment (mahoneycm) [inline _data/changelogs/component-date-picker.yml:7]: **Suggestion (non-blocking):** Ok sorry to double back to this but having second thoughts about "cannot be interacted with. Considering something like:

```suggestion
    summaryAdditional: Now, the input element and calendar icon remain discoverable to screen readers but prevent the user from selecting a date.
```

CC: @amyleadem cus she's good at these... 😁

**Comment by mahoneycm** [comment]
> Hi, Rachel!
> 
> A couple of little things:
> 
> In the title, change "Chanlgelog to Changelog" because I'm being nitpicky. ;)
> 
> How about this for wording of summaryAdditional? "Now, the input element and calendar icon remain discoverable to screen readers but users are prevented from selecting a date."

RE: PR Title change, we've been following the following pattern for changelog entries:

> USWDS-Site: _Quick description of work_ [#NNNN _Contribution PR nubmer_]

so something like...

> USWDS-Site: Add changelog for date picker aria-disabled updates [#6013]

Open to suggestions if you or @cathybaptista have any!

File: _data/changelogs/component-date-picker.yml (hunk 4 above)
Comment (RachelCorsino) [reply _data/changelogs/component-date-picker.yml]: Resolved in [63ee9fc](https://github.com/uswds/uswds-site/pull/2794/commits/63ee9fccd2f039087be6d17a0450ffc4caa19d46). Made a small update to what you recommended @mahoneycm so let me know what you think!

File: _data/changelogs/component-date-picker.yml [hunk 5]
```diff
@@ -2,6 +2,14 @@ title: Date picker
 type: component
 changelogURL:
 items:
+  - date: NNNN-NN-NN
+    summary: Enhanced `aria-disabled` behaviors to match `disabled` behaviors.
+    summaryAdditional: Now, the input element and calendar icon remain discoverable to screen readers but prevent the user from selecting or inputting a date.
```
Comment (amyleadem) [inline _data/changelogs/component-date-picker.yml:7]: **suggestion:** Recommend we make this summary simpler to focus more on what the user will care about. 

```suggestion
    summary: Added `aria-disabled` to the list of expected attributes. 
    summaryAdditional: Now, the component will disable toggle when the `aria-disabled` attribute is present.
```

File: _data/changelogs/component-date-picker.yml
Comment (amyleadem) [inline _data/changelogs/component-date-picker.yml:1]: **Suggestion:** Does this change affect date range picker as well? If so, we should consider adding a changelog in component-date-range-picker.yml as well.

File: _data/changelogs/component-date-picker.yml (hunk 5 above)
Comment (RachelCorsino) [reply _data/changelogs/component-date-picker.yml]: Resolved in [a591638](https://github.com/uswds/uswds-site/pull/2794/commits/a59163890d9f3e277ca43f34a3dfd72c3d878a2b)

File: _data/changelogs/component-date-picker.yml
Comment (RachelCorsino) [reply _data/changelogs/component-date-picker.yml:1]: Yes! It does... I'll go ahead and add a changelog for date range picker as well and create a PR for it.

**Review by amyleadem — APPROVED** [review]
LGTM!
